test/test_torch.py

```diff
@@ -16448,16 +16447,160 @@ def test_svd_tall_some_col_maj(self, device, dtype):
+_ufunc_meta = collections.namedtuple('_ufunc_meta',
+                                     'is_floating alt_impl complex_on complex_out')
+def ufunc_meta(is_floating=False, alt_impl=None, complex_on=('cpu', 'cuda'), complex_out=('cpu', 'cuda')):
+    return _ufunc_meta(is_floating, alt_impl, complex_on, complex_out)
+
+# Dict of unary ufuncs and their associated test metadata
+unary_ufuncs = {
+    'sin': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'cos': ufunc_meta(is_floating=True, complex_on=('cpu'), complex_out=('cpu')),
+    'tan': ufunc_meta(is_floating=True, complex_on=('cpu'), complex_out=('cpu')),
+    'asin': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'acos': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'atan': ufunc_meta(is_floating=True, complex_on=('cpu'), complex_out=('cpu')),
+    'sinh': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'cosh': ufunc_meta(is_floating=True, complex_on=('cpu'), complex_out=('cpu')),
+    'tanh': ufunc_meta(is_floating=True, complex_on=('cpu'), complex_out=('cpu')),
+    'ceil': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'floor': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'exp': ufunc_meta(is_floating=True, complex_on=('cpu'), complex_out=('cpu')),
+    'expm1': ufunc_meta(is_floating=True, complex_on=(), complex_out=()),
+    'log': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'log10': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'log1p': ufunc_meta(is_floating=True, complex_on=(), complex_out=()),
+    'log2': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'sqrt': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'rsqrt': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'trunc': ufunc_meta(is_floating=True, complex_on=('cpu')),
+    'erf': ufunc_meta(is_floating=True, complex_on=(), complex_out=()),
+    'erfc': ufunc_meta(is_floating=True, complex_on=(), complex_out=()),
+    'erfinv': ufunc_meta(is_floating=True, complex_on=(), complex_out=()),
+    'lgamma': ufunc_meta(is_floating=True, complex_on=(), complex_out=()),
+    'digamma': ufunc_meta(is_floating=True, complex_on=(), complex_out=()),
+    'sigmoid': ufunc_meta(is_floating=True, complex_on=('cpu'), complex_out=('cpu'))
+}
+
+# Creates and adds a test of the given unary floating ufunc's type promotion
+# Unary floating ufuncs promote tensors of bool and integral types to
+# the default scalar type.
+def generate_unary_floating_ufunc_promo_test(cls, op_str):
+    int_types = [torch.bool, torch.uint8, torch.int8, torch.int16,
+                 torch.int, torch.long]
+    float_types = [torch.float, torch.double]
+    float_types_cuda = [torch.half, torch.float, torch.double]
+    complex_types = [torch.complex64, torch.complex128]
+    default_types = [torch.float, torch.double]
+    default_types_cuda = [torch.half, torch.float, torch.double]
+
+    def test_fn(self, device):
+        op = getattr(torch, op_str)
```

Comment:
We should test the methods, too: 

`meth = getattr(torch.Tensor, op_str)`

Reply:
Agreed. This is solved in the recent commit: https://github.com/pytorch/pytorch/pull/33322/commits/29ae207f362efe4040b598355cfe1db1fadb82ad. Thanks!